src/server/dfly_main.cc

```diff
@@ -132,6 +134,22 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   return true;
 }
 
+bool CreatePidFile(const string& path) {
+    ofstream ofs;
+    ofs.exceptions(ios::failbit | ios::badbit);
+
+    try {
+      ofs.open(path, ios::trunc);
+      ofs << getpid();
+      ofs.close();
+    } catch (ios::failure& ex) {
```

Comment:
I also do not use c++ exceptions. These are poorly designed in c++ and you can see that I always return errors explicitly.